packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -686,4 +693,120 @@ export class AbstractQueryInterfaceTypeScript {
   ): Promise<void> {
     await this.sequelize.queryRaw(this.queryGenerator.getToggleForeignKeyChecksQuery(enable), options);
   }
+
+  /**
+   * Create a new savepoint
+   *
+   * @param transaction
+   * @param options
+   */
+  async createSavepoint(transaction: Transaction, options: CreateSavepointOptions): Promise<void> {
```

Comment:
all query interface methods related to transactions should be marked as private. Maybe the name should start with `_` or `_unsafe_`, or similar. People should not use them directly, they should only be used via the transaction object

Ideally, they should be moved to the Internal version of QueryInterface, but I don't know how to expose it to the Transaction class yet. Maybe via an internal symbol?

Reply:
I've added a note to each of the functions about this.
Marking the methods as private will prevent the use of them inside the transaction class.

Other private/internal methods that need to be accessible outside their classes currently just prefix `_` to the method name, shall I do this to keep it consistent?

Reply:
Yep! That's what I meant by marking them as private, prefixing them with `_` is a good enough indicator that users should not attempt to use it themselves :)

Reply:
Done